### ISSUE TITLE
Do not zoom/pan to first Nominatim result if other results are equally probable

### ISSUE THREAD

**Issue by Soundlover99** [openstreetmap/openstreetmap-website#3991]
### Problem

If Nominatim finds several results (e.g., search term / address is ambiguous) which are equally probable or plausible, then the current behaviour of zooming/panning on the first result creates the false impression that this is the "best" or "correct" result. In my experience, users tend to trust search engines that the first result is the best / desired result, they rarely check the following results. In case of e.g. duplicate street names within the same city / zip / postal code, all results are equal as long as the district / village etc. (e.g. Tag: addr:suburb) has not been specified or selected. See "current behaviour" screenshot.

Yes, the city administrations should resolve such address conflicts, but they don't, becaus this would be even more hassle than leaving it the way it is.

Same goes for searching generic points of interest in a city; zooming in on the "first" result, and subsequently showing only one red position tag (corresponding to the result on which the cursor hovers) at a time is rarely the desired behaviour.

More elaborate background can be found here: https://help.openstreetmap.org/questions/74796/nominatim-search-results-prioritization

### Description

- If a query leads to several "equal" results, the map view should not zoom/pan on any one of them
- "Equal" results can be:
     - If the first Nominatim search result has the same address rank as one or more of the subsequent results, or
     - Results which fully match the search term (especially if the city/municipality is included) but require an additional address part (e.g. Tag: addr:suburb) not contained in the query for further narrowing down
- All such "equal" results should be displayed simultaneously by an equivalent number of red position tags, on a map suitably zoomed/panned to show all of these (similar to "show all hardware stores in Berlin"). This way, the user is immediately aware that there are several equally probable results to his/her query. Hovering over the results list enlarges the corresponding position tag.  See "Mockup" screenshot.
- This behaviour could extend not only to residential addresses, but also to points of interest of all kind, e.g. "Supermarket Berlin" or "Museum Paris".

### Screenshots

![OSM_current_behaviour_1](https://user-images.githubusercontent.com/129313036/228653260-8c36afe0-c30e-464b-8830-04a006bf2355.png)
![Mockup_equal_search_results](https://user-images.githubusercontent.com/129313036/228653285-da6d7744-bd9f-47bd-9158-ff9846ff6aa3.png)

**Comment by tomhughes** [comment]
As far as I know there is no concept of equal results - nominatim orders the results and we display them in the order it sends them to us.

**Comment by tomhughes** [comment]
Certainly we don't want to be having to do a lot of complicated analysis of the results that nominatim sends - it's the geocode not us and we don't want to be second guessing it.

So if there's some simple value in the nominatim result we can compare then maybe we can consider this but I don't think we're going to start applying a whole series of rules like you propose at the client side.

**Comment by lonvia** [comment]
Nominatim does export the place importance but that only tells you about the absolute well-knownness of a result, not about how well it fits the query. It would be close to useless in an address search like in the example above.  So, no, there is no way of doing this right now.

**Comment by Soundlover99** [comment]
Thanks everyone. Since no one objected to the issue / feature request itself (from a user point of view), I assume that you agree with the request in general?

I understand that there is no immediate technical solution. Being a techie myself, I know this point of view very well. But let's look at it from the end: OSM has strong corporate competition, as we all know, and if OSM lags behind in basic usability, and isn't better in some practical aspects, I don't see it catching up, which is a real shame seeing the tons of volunteer work going into OSM. Wikipedia would be a good example where good usability supports widespread adoption (or at least doesn't hinder it).

And the handling of multiple search results is IMHO an important part of basic usability, not a fancy cherry-on-top. Users want quick but correct answers: Where can I find what I need right now? Where are stores, hospitals, bus stops, museums, restaurants in my vicinity? Or: Is the address shown to me really the one I'm looking for, or do others fit my query just as well?

If we agree on this, then the next logical question is: How could we get there? Probably by posting a feature request to Nominatim, but I'd like to avoid getting the response "there is no need and no defined interface with OSM for this, no can do" ... any hints how to proceed best?

**Comment by tomhughes** [comment]
Right, but this web site's design goal is to support mappers, not end users. It is specifically not our goal to provide a Google Maps replacement to end users.
